### ISSUE TITLE
How to enable url hash mode?

### ISSUE THREAD

**Issue by pkubangbang** [vitejs/vite-plugin-react-pages#46]
Hello!

First, BIG THANKS for all your hard work on this great project! I think it deserves applause for a competitive solution in the React world!

Just caught it up in the previous version of the `doc-site` code that **a `useHashRouter: true` is exposed in the `vite-plugin-react-pages` configs**, like:

```
// vite.config.ts, modified from "basic" demo.

import { defineConfig } from 'vite'
import * as path from 'path'
import reactRefresh from '@vitejs/plugin-react-refresh'
import mdx from 'vite-plugin-mdx'
import pages from 'vite-plugin-react-pages'

module.exports = defineConfig({
  plugins: [
    reactRefresh(),
    mdx(),
    pages({
      useHashRouter: true, // <--- here it is
      pagesDir: path.join(__dirname, 'pages'),
    }),
  ],
})

```

Question: 
1. Is it ready to use?
2. If so, will it affect the pageStrategy?

(I've found that when changed, my inter-links within markdown files will have to use something like `[another-page](#/another-page)` with a # in there.)

**Comment by csr632** [comment]
Thanks for your encouragement!

`useHashRouter` is ready to use. I doesn't affect the pageStrategy. The reason that I drop `useHashRouter: true` in the `doc-site` project is: `useHashRouter: true` is not suitable for static site generation. Because the route path is after `#`, the first HTML request will always be like `https://vitejs.github.io/vite-plugin-react-pages/`. Event when you are loading `https://vitejs.github.io/vite-plugin-react-pages/#/getting-stated`, you will get the ssr-rendered `/index.html` page instead of the `/getting-stated.html`. So we lose the benefit of static site generation. It is the inherent weakness of hash router.

> (I've found that when changed, my inter-links within markdown files will have to use something like [another-page](#/another-page) with a # in there.)

Yeah, that is a flaw of hash router mode. Currently you can work around it by manually render `<Link>` jsx, [like how doc-site do it](https://raw.githubusercontent.com/vitejs/vite-plugin-react-pages/d9c16f9aa54aa7e6e58c636ff8f967cb7d73c909/doc-site/pages/fs-routing%24.mdx).
I will fix it by introduce a mdx plugin that transform `[link](/app-path)` into `<Link to="/app-path">link</Link>` automatically.

**Comment by csr632** [comment]
> I will fix it by introduce a mdx plugin that transform [link](/app-path) into <Link to="/app-path">link</Link> automatically.

This is fixed in `vite-pages-theme-doc@3.1.3`.
